### ISSUE TITLE
MeiliSearchApiError. Error code: database_size_limit_reached.

### ISSUE THREAD

**Issue by Simanas** [meilisearch/meilisearch#3920]
With all of my respect to the great work that has been done on this project, issue that I want to report has been already reported multiple times and I find it quiet dumb that it still exists, compared to the great work that has already been done. I also think it requires immediate fix as if somebody wants to use this for serious amounts of load, than it is really not yet ready for it. However I see great potential of this engine becoming the best search engine in a very near future, but for that to happen it must be ready for the big data!

We have a lot of changing data that is being re-indexed on a daily basis. Our first honey moon with meilisearch went perfect in every possible way, however few days ago she got completely stuck without any warning and now it seems that the only way to fix this is to dump our entire data folder and start our relationships all over again (which we already did yesterday and got in to the same issue again).

So now whenever we try to pass any new task to her, she is throwing this error:

"MeiliSearchApiError: MeiliSearchApiError. Error code: database_size_limit_reached. Error message: MDB_MAP_FULL: Environment mapsize limit reached Error documentation: https://docs.meilisearch.com/errors#database_size_limit_reached Error type: internal"

Since we started all over again by deleting our old indexes, new indexes are not that big yet, takes up ~900 Mb in total.

However I have figured out that tasks folder is at 10GB as it got a lot of tasks to do over the first month of use (all in batches). If I get it correctly all of those tasks are successfully executed, however they are still there, taking up the space, but most importantly it got reached some hard coded 10GB limit (where do we change that limit?). So we have learned the hard way that we must do tasks cleanup ourselves.... that's not really a problem, but if this is the way it works it should be documented on installation and use documentation in red bold text!

Ok, so we got stuck... well let's try and clear some of those tasks up... We use Python, therefore we do: 

`ms_client.delete_tasks(parameters={'statuses': ['succeeded']})`

Good try! Same error! It wont accept a single new task even if it's dedicated to clean her self up... So now how do we go about this? Is there any setting that we can change to increase that 10GB limit?

Now... quick suggestion for an easy fix: Having a limit on tasks queue is a good idea, however users should have an option of changing how much space it takes up (10Gb is a good default value), but most importantly, this task queue should clean it self up on a very regular basis, so that instead of throwing an error it should drop oldest tasks to make up a space for new ones.

**Comment by oluademola** [comment]
Hello @Simanas. Sorry for the inconvenience. What version of meilisearch are you using? In the older version of the engine, reaching the maximum size (10GB) for the task database is indeed the case. However, we have addressed this problem in the latest version by introducing an automatic task deletion feature, as mentioned on the [release page](https://github.com/meilisearch/meilisearch/releases/tag/v1.2.0).

To resolve the issue and unblock the situation, as far as I know, I am afraid you will need to start fresh since it does not take any tasks into the queue to free up the needed space. 

cc @meilisearch/engine-team, perhaps they have alternative suggestions to help unblock it.

**Comment by Simanas** [comment]
O wow! Yes we are on v1.0.2.

These improvements mentioned on v1.2.0 release page really hits the eyeball! Great work guys! Closing the issue and moving up to v1.2.0! :) Sorry for not finding this out by my self. However this might help some other lost older version users to find the proper answer! :)

Brilliant work that you do here! Thank you and bless you all!
